ui/signals.py

```diff
@@ -46,18 +48,37 @@ def delete_youtube_video(sender, **kwargs):
         remove_youtube_video.delay(kwargs['instance'].id)
 
 
-@receiver(post_save, sender=Video)
-def sync_youtube(sender, **kwargs):
+@receiver(post_save, sender=Collection)
+def update_collection_youtube(sender, **kwargs):
     """
+    If a collection's stream source is changed, sync YoutubeVideo objects for public Videos
+    """
+    if settings.ENABLE_VIDEO_PERMISSIONS:
+        for video in kwargs['instance'].videos.filter(is_public=True):
```

Comment:
Wouldn't we want to do all videos and not just the public ones? From the docstring it looks like `sync_youtube` should delete videos which aren't public

Reply:
Only public videos should potentially be on Youtube, and any videos that are changed from public to private should be deleted from Youtube via the `update_video_youtube` signal.  I was filtering here to avoid a little overhead in iterating through all the private videos, but I can remove this just in case there's any private videos that somehow get a Youtube video associated with them, maybe if the signal fails.